brave/browser-laptop
# Fixes add funds dialog

## Submitter Checklist:


Resolves #12078

Auditors:

also added unit test for it

## Test Plan:

steps based on comment: https://github.com/brave/browser-laptop/issues/12078#issuecomment-348343208

1) Clean profile
2) start browser with staging server
```
LEDGER_ENVIRONMENT=staging npm run start
```
3) Enable payments
4) Visit some sites so that you have a few ledger entries showing on the Payments tab
5) Fund wallet (ex: claim UGP grant)
6) Close Brave
7) Update `reconcileStamp` in `ledger-state.json` to the epoch for now (or a time in the past). ([This site](https://www.freeformatter.com/epoch-timestamp-to-date-converter.html) is handy for conversions)
8) Launch Brave again using
```
LEDGER_NO_FUZZING=true LEDGER_VERBOSE=true LEDGER_ENVIRONMENT=staging npm run start
```
9) Wait for contribution to be made
10) "Needs deposit" notice should NOT show

## Reviewer Checklist:

Tests
